Backend/app/kafka-consumers/src/Consumer/PersonStats/Processor.hs

```diff
@@ -0,0 +1,76 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Consumer.PersonStats.Processor
+  ( updatePersonStats,
+  )
+where
+
+import Data.Time
+import "rider-app" Domain.Action.UI.CleverTap as DC
+import "rider-app" Domain.Types.BookingCancellationReason as SBCR
+import "dynamic-offer-driver-app" Domain.Types.Ride as DDR
+import Environment
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Kernel.Utils.Logging
+import "sessionizer-metrics" Lib.SessionizerMetrics.Types.Event as E
+import "rider-app" Storage.Queries.BookingCancellationReason as QBCR
+import "rider-app" Storage.Queries.Person.PersonStats as QP
+import "rider-app" Storage.Queries.Ride as QR
+import "rider-app" Tools.Error
+import "dynamic-offer-driver-app" Tools.Event as TE
+
+updatePersonStats :: E.Event TE.Payload -> Text -> Flow ()
+updatePersonStats event _ = do
+  when (isJust event.personId && event.service == RIDER_APP) do
+    let personId = Id (fromMaybe "" event.personId)
+
+    personStats_ <- Esq.runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+    when (personStats_.userCancelledRides == 0 && personStats_.completedRides == 0 && personStats_.weekendRides == 0 && personStats_.weekdayRides == 0 && personStats_.offPeakRides == 0 && personStats_.eveningPeakRides == 0 && personStats_.morningPeakRides == 0 && personStats_.weekendPeakRides == 0) $ do
```

Comment:
Also can be written as `all (== 0) [personStats_.userCancelledRides, personStats_.completedRides, personStats_.weekendRides, personStats_.weekdayRides, personStats_.offPeakRides, personStats_.eveningPeakRides, personStats_.morningPeakRides, personStats_.weekendPeakRides]`